lib/DAV/Auth/Backend/IMAP.php

```
@@ -0,0 +1,79 @@
+<?php
+
+namespace Sabre\DAV\Auth\Backend;
+
+/**
+ * This is an authentication backend that uses imap.
+ *
+ * @copyright Copyright (C) fruux GmbH (https://fruux.com/)
+ * @author Michael Niewöhner (foss@mniewoehner.de)
+ * @author rosali (https://github.com/rosali)
+ * @author Evert Pot (http://evertpot.com/)
+ * @license http://sabre.io/license/ Modified BSD License
+ */
+class IMAP extends AbstractBasic {
+
+    /**
+     * IMAP server and options
+     *
+     * @var string
+     */
+    protected $imap;
+
+    /**
+     * Creates the backend object.
+     *
+     * @param string $imap
+     */
+    function __construct($imap_host, $imap_flags = '') {
+
+        $this->imap_host = $imap_host;
+        $this->imap_flags = $imap_flags;
+
+    }
+
+    /**
+     * Connects to an IMAP server and tries to authenticate
+     *
+     * @param string $username
+     * @param string $password
+     * @return bool
+     */
+    protected function imapOpen($username, $password) {
+
+        $mailbox = "{" . $this->imap_host . $this->$imap_flags . "/readonly}";
+        $success = false;
+
+        try {
+            $imap = imap_open($mailbox, $username, $password, 0, 1);
+
+            if ($imap) {
+                imap_close($imap);
+                $success = true;
+            } else {
+              $success = false;
+            }
+        } catch (\Exception $e) {
```

Oh. Just saw that, too ;-)
"* Mapping PHP errors to exceptions." .....

I don't like the idea of suppressing errors with the @ op. Catching the exception and logging it is the best solution I think